src/transaction/log_page_buffer.c

```diff
@@ -1123,123 +771,86 @@ logpb_fix_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, PAGE_FETCH_MODE fetc
   assert ((fetch_mode == NEW_PAGE) || (fetch_mode == OLD_PAGE));
   assert (LOG_CS_OWN_WRITE_MODE (thread_p));
 
-  START_EXCLUSIVE_ACCESS_LOG_PB (rv, thread_p);
-
-  log_bufptr = (LOG_BUFFER *) mht_get (log_Pb.ht, &pageid);
-  if (log_bufptr == NULL)
+  if (pageid == LOGPB_HEADER_PAGE_ID)
     {
-      /* 
-       * Page is not resident. Find a replacement buffer for it
-       */
-
-      stat_page_found = PERF_PAGE_MODE_OLD_LOCK_WAIT;
-
-      while (1)
+      log_bufptr = &log_Pb.header_buffer;
+    }
+  else
+    {
+      index = logpb_get_log_buffer_index ((int) pageid);
+      if (index >= 0 && index < log_Pb.num_buffers)
 	{
-	  log_bufptr = logpb_replace (thread_p, &retry);
-	  if (log_bufptr == NULL)
-	    {
-	      if (retry)
-		{
-		  continue;
-		}
-	      else
-		{
-		  goto error;
-		}
-	    }
-	  else
-	    {
-	      break;
-	    }
+	  log_bufptr = &log_Pb.buffers[index];
+	}
+      else
+	{
+	  er_set (ER_FATAL_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_PAGE_CORRUPTED, 1, pageid);
+	  return NULL;
 	}
 
-      assert (log_bufptr != NULL);
+    }
+  assert (log_bufptr != NULL);
 
-      /* 
-       * Remove the current page from the hash table and get the desired
-       * page from disk when the page is an old page
-       */
-      if (log_bufptr->pageid != NULL_PAGEID)
+  if (log_bufptr->pageid != NULL_PAGEID && log_bufptr->pageid != pageid)
+    {
+      if (log_bufptr->dirty == true)
 	{
-	  (void) mht_rem (log_Pb.ht, &log_bufptr->pageid, NULL, NULL);
+	  if (logpb_write_page_to_disk (thread_p, log_bufptr->logpage, log_bufptr->pageid) != NO_ERROR)
+	    {
+	      assert_release (false);
+	      return NULL;
+	    }
```

Comment:
@funkydvd 

sorry I'd like to change my proposal. 
Please ignore the above comment and add it to `logpb_write_page_to_disk`. I'm going to also leave a comment of the function.